interfaces/sourcegen/sourcegen/_TagFileParser.py

```diff
@@ -0,0 +1,296 @@
+
+    @classmethod
+    def from_xml(cls, qualified_name, xml):
+        """Create tag information based on XML data."""
+        base = ""
+        if "::" in qualified_name:
+            base = qualified_name.split("::", 1)[0]
+
+        xml_tree = ET.fromstring(xml)
+        return cls(base,
+                   xml_tree.find('type').text,
+                   xml_tree.find('name').text,
+                   xml_tree.find('arglist').text,
+                   xml_tree.find('anchorfile').text.replace(".html", ".xml"),
+                   xml_tree.find('anchor').text)
+
+    def __bool__(self):
+        return all([self.type, self.name, self.arglist, self.anchorfile, self.anchor])
+
+    @property
+    def signature(self):
+        """Generate function signature based on tag information."""
+        return f"{self.type} {self.name}{self.arglist}"
+
+    @property
+    def id(self):
+        """Generate doxygen id."""
+        return f"{self.anchorfile.replace('.xml', '')}_1{self.anchor}"
+
+    @property
+    def qualified_name(self):
+        """Return qualified name."""
+        if self.base:
+            return f"{self.base}::{self.name}"
+        return self.name
+
+
+@dataclass(frozen=True)
+@with_unpack_iter
+class TagDetails(TagInfo):
+    """Create tag information based on XML data."""
+
+    location: str = ""  #: File containing doxygen description
+    briefdescription: str = ""  #: Brief doxygen description
+    parameterlist: list[Param] = None  #: Annotated doxygen parameter list
+
+
+class TagFileParser:
+    """Class handling contents of doxygen tag file."""
+
+    _known: dict[str, str]  #: Dictionary of known functions and corresponding XML tags
+
+    def __init__(self, bases: dict[str, str]) -> None:
+        tag_file = _TAG_PATH / "Cantera.tag"
+        if not tag_file.exists():
+            msg = (f"Tag file does not exist at expected location:\n    {tag_file}\n"
+                "Run 'scons doxygen' to generate.")
+            _LOGGER.critical(msg)
+            sys.exit(1)
+
+        with tag_file.open() as fid:
+            doxygen_tags = fid.read()
+
+        logging.info("Parsing doxygen tags...")
+        self._parse_doxyfile(doxygen_tags, bases)
+
+    def _parse_doxyfile(self, doxygen_tags: str, bases: list[str]) -> None:
+        """Retrieve class and function information from Cantera namespace."""
+
+        def xml_compounds(kind: str, names: list[str]) -> dict[str,str]:
+            regex = re.compile(rf'<compound kind="{kind}"[\s\S]*?</compound>')
+            found = []
+            compounds = {}
+            for compound in re.findall(regex, doxygen_tags):
+                qualified_name = ET.fromstring(compound).find("name").text
+                compound_name = qualified_name.split(":")[-1]
+                if compound_name in names:
+                    found.append(compound_name)
+                    compounds[compound_name] = compound
+                    if not set(names) - set(found):
```

Comment:
I don't think this is strictly necessary but IMO improves clarity:
```suggestion
                    if not (set(names) - set(found)):
```